Add missing entries to default.json

Luc45 opened the pull request:
### All Submissions:

* [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR updates the file `packages/js/e2e-environment/config/default.json` with new keys that were added on `plugins/woocommerce/tests/e2e/config/default.json` - and only the ones needed for the tests to run.

Closes #31454

### How to test the changes in this Pull Request:

1. Temporarily move/delete/rename file `plugins/woocommerce/tests/e2e/config/default.json`
3. Run e2e tests `pnpm nx test-e2e woocommerce`
4. Assert that the file `packages/js/e2e-environment/config/default.json` has the same contents as `packages/js/e2e-environment/config/default/default.json`
5. Assert that the tests complete without errors

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

Luc45 committed:
Add missing entries to default.json

Luc45 committed:
Delete bk file

Luc45 commented:
@rrennick  All tests run, except for one right at the beginning related to "Analytics": `plugins/woocommerce/tests/e2e/specs/admin-analytics/analytics-overview.test.js`

This test fails even with the original `plugins/woocommerce/tests/e2e/config/default.json` file. Is there anything I should do in this PR about that?

![image](https://user-images.githubusercontent.com/9341686/147363523-cd431a2a-e052-4ea8-b4b4-84aaf17aae81.png)

rrennick commented:
> This test fails even with the original plugins/woocommerce/tests/e2e/config/default.json file. Is there anything I should do in this PR about that?

We have an issue for this happening locally: https://github.com/woocommerce/woocommerce/issues/30581. But, in general we try to address one issue per PR and would still address it in a separate PR.

rrennick commented:
@Luc45 Nice work on this. This tested great for me locally. There are a few things that will need to be addressed:

I forgot to mention that each npm package has its own changelog. For npm package changes, the change entries go in the package changelogs  vs the PR description. Can you add an entry to the [e2e-environment changelog](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/e2e-environment/CHANGELOG.md) and remove the one from your PR description?

> 2. Optionally, delete packages/js/e2e-environment/config/default.json

Changes to the npm packages need to work as a package. Testing changes to the packages should not involve the tester making changes to any files under `packages/js`.

> The only JSON array that wasn't necessary from the original default.json was:

Nice catch. While the tests do work without these entries, not having the entries changes [what is being tested](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/e2e-core-tests/specs/api/variable-product.test.js#L43:L49). So, we should include those entries.

@zhongruige I'll be handing this over for a follow up review.

Luc45 committed:
Add "defaultAttributes" JSON key

Luc45 committed:
Merge branch 'trunk' into e2e/add-missing-entries-to-default-json

Luc45 committed:
Add changelog

Luc45 commented:
> @Luc45 Nice work on this. This tested great for me locally. There are a few things that will need to be addressed:
> 
> I forgot to mention that each npm package has its own changelog. For npm package changes, the change entries go in the package changelogs vs the PR description. Can you add an entry to the [e2e-environment changelog](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/e2e-environment/CHANGELOG.md) and remove the one from your PR description?
> 
> > 2. Optionally, delete packages/js/e2e-environment/config/default.json
> 
> Changes to the npm packages need to work as a package. Testing changes to the packages should not involve the tester making changes to any files under `packages/js`.
> 
> > The only JSON array that wasn't necessary from the original default.json was:
> 
> Nice catch. While the tests do work without these entries, not having the entries changes [what is being tested](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/e2e-core-tests/specs/api/variable-product.test.js#L43:L49). So, we should include those entries.
> 
> @zhongruige I'll be handing this over for a follow up review.

- Added "defaultAttributes" JSON key
- Added changelog
- Tweaked test steps in PR description

zhongruige commented on packages/js/e2e-environment/config/default/default.json:
```diff
@@ -13,16 +13,137 @@
+			"option": "Blue"
+		  }
+		]
+	  }
```
Looks like the formatting is a bit off here, with many of the brackets on the same line:

<img width="211" alt="Screen Shot 2021-12-27 at 11 38 02" src="https://user-images.githubusercontent.com/71906536/147498770-40255abe-dbfc-4e39-871b-d6013f6157f6.png">

It looks like the file itself is a mix of spaces and tabs, causing some rendering issues in GitHub while it looks fine in the IDE I am using. If we can, since we're changing this file, let's see about updating the file to use tabs matching the [WordPress PHP coding standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#indentation).

zhongruige reviewed (commented):
Changes look good! I ran these tests locally and everything passed for me. One comment on updating the `default.json` file formatting for consistency in indentation for review.

Also as these changes are being done in the `e2e-environment` package, I added that label so it's easier to track where the changes are being done.

Luc45 committed:
Normalized 4 spaces as tabs

zhongruige reviewed (approved):
Looks good--thank you for getting in these changes @Luc45!
